addon/chart-data-updater.js

```diff
@@ -37,20 +37,36 @@ export default Ember.Object.extend({
     var chart = this.get('chart');
     var needUpdate = false;
 
-    data.forEach(function(segment, i) {
-      if (typeof chart.segments[i] !== 'undefined') {
+    data.forEach(function(segment) {
+      var currentSegment = chart.segments.findBy('label', segment.label);
+      if(chart.segments.findBy('label', segment.label)) {
         segment.value = segment.value || 0;
-        if (chart.segments[i].value !== segment.value) {
-          chart.segments[i].value = segment.value;
+        if (currentSegment.value !== segment.value) {
```

Comment:
The reason this if was made, is because of the needUpdate variable.

I think the reason behind this `needUpdate variable` is the following: if you update your graph, and by change the new data is equal to the old data, there is no need to update the chart.